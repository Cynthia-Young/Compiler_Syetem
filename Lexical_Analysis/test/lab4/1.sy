int a;

int main()
{
    int a;
    a = 1 + 2;
    if(a >= 5)
        return 1;
    return 0;
    a = 0123;  
    a = 0x123a; 
    a= 0X123A; 
    
    /*
    
    if(a < 5)
        return 1;
    return 0;
    
    */
}