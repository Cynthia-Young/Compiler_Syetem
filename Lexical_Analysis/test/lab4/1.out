[DEBUG LAB4]: 	INT	int	1
[DEBUG LAB4]: 	ID	a	1	0x562f50dafec0
[DEBUG LAB4]: 	SEMICOLON	;	1
[DEBUG LAB4]: 	INT	int	3
[DEBUG LAB4]: 	ID	main	3	0x562f50daff24
[DEBUG LAB4]: 	LPAREN	(	3
[DEBUG LAB4]: 	RPAREN	)	3
[DEBUG LAB4]: 	LBRACE	{	4
[DEBUG LAB4]: 	INT	int	5
[DEBUG LAB4]: 	ID	a	5	0x562f50db8b30
[DEBUG LAB4]: 	SEMICOLON	;	5
[DEBUG LAB4]: 	ID	a	6	0x562f50db8b30
[DEBUG LAB4]: 	ASSIGN	=	6
[DEBUG LAB4]: 	DECIMIAL	1	6
[DEBUG LAB4]: 	ADD	+	6
[DEBUG LAB4]: 	DECIMIAL	2	6
[DEBUG LAB4]: 	SEMICOLON	;	6
[DEBUG LAB4]: 	IF	if	7
[DEBUG LAB4]: 	LPAREN	(	7
[DEBUG LAB4]: 	ID	a	7	0x562f50db8b30
[DEBUG LAB4]: 	GREATERorEQUAL	>=	7
[DEBUG LAB4]: 	DECIMIAL	5	7
[DEBUG LAB4]: 	RPAREN	)	7
[DEBUG LAB4]: 	RETURN	return	8
[DEBUG LAB4]: 	DECIMIAL	1	8
[DEBUG LAB4]: 	SEMICOLON	;	8
[DEBUG LAB4]: 	RETURN	return	9
[DEBUG LAB4]: 	DECIMIAL	0	9
[DEBUG LAB4]: 	SEMICOLON	;	9
[DEBUG LAB4]: 	ID	a	10	0x562f50db8b30
[DEBUG LAB4]: 	ASSIGN	=	10
[DEBUG LAB4]: 	OCT	83	10
[DEBUG LAB4]: 	SEMICOLON	;	10
[DEBUG LAB4]: 	ID	a	11	0x562f50db8b30
[DEBUG LAB4]: 	ASSIGN	=	11
[DEBUG LAB4]: 	HEX	4666	11
[DEBUG LAB4]: 	SEMICOLON	;	11
[DEBUG LAB4]: 	ID	a	12	0x562f50db8b30
[DEBUG LAB4]: 	ASSIGN	=	12
[DEBUG LAB4]: 	HEX	4666	12
[DEBUG LAB4]: 	SEMICOLON	;	12
[DEBUG LAB4]: 	RBRACE	}	21
